core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -941,6 +965,12 @@ private[spark] object SparkSubmitUtils {
       val ri = ModuleRevisionId.newInstance(mvn.groupId, mvn.artifactId, mvn.version)
       val dd = new DefaultDependencyDescriptor(ri, false, false)
       dd.addDependencyConfiguration(ivyConfName, ivyConfName + "(runtime)")
+      if (mvn.classifier.isDefined) {
```

Comment:
I have a feeling we might be missing it when we copy the files from the `local repo` to the ivy cache. The way the resolution works is as follows:

 1. Ivy searches local cache
 2. Ivy searches remote repos
 3. Ivy downloads from remote repo to local repo
 4. Ivy copies the file from local repo to cache

Then the files are served from the local cache. Step 4 is called `retrieval` and happens on line 1159. I think that's what you're missing